src/python/pants/option/global_options.py

```diff
@@ -183,14 +187,21 @@ def from_options(cls, options: Options) -> ExecutionOptions:
             else:
                 remote_execution_headers = auth_plugin_result.execution_headers
                 remote_store_headers = auth_plugin_result.store_headers
+                if remote_instance_name != auth_plugin_result.instance_name:
```

Comment:
This logic still seems wrong to me. For example, if `remote_instance_name` is set to `main` and `auth_plugin_result.instance_name` is `None`, then `remote_instance_name != auth_plugin_result.instance_name` and `remote_instance_name` will be overwritten to `None`, which is not what should happen.

Reply:
Ah! You are totally right. Thanks Tom!